### PR TITLE
feat: Extract client-side stack traces from Minidumps (NATIVE-195)

### PR DESCRIPTION
**Pull request by Swatinem**
This adds support for reading our custom minidump extension, and transforming it into `types::RawStacktrace` which we will use in a followup PR to combine with the stacktrace that was extracted via breakpad.

The PR also contains a documentation of the binary format of our extension.

#skip-changelog

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -3693,6 +3774,15 @@ dependencies = [
  "winapi 0.3.9",
 ]
 
+[[package]]
+name = "test-assembler"
+version = "0.1.5"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "5ccc488d8ed3a4a31f68372ba429642245ddbdf6642054334b7fcc06264cbde9"
+dependencies = [
+ "byteorder 0.5.3",
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply Cargo.lock:3197]: lol, well I guess you gotta motivate people to release more frequently, and keep their dependencies up to date. But oh well, we are not too good examples for that ourselves :-(

File: crates/symbolicator/src/services/minidump.rs [hunk 2]
```diff
@@ -0,0 +1,858 @@
...
+
+//! The extension is a minidump stream with the ID `0x53790001`.
+//!
+//! The format comprises:
+//! - a header
+//! - a list of threads
+//! - a list of frames
+//! - a symbol data section
+//!
+//! The data is written in native endianness and can currently only be parsed as little endian.
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator/src/services/minidump.rs:11]: Maybe rephrase:
The extension bytes are being written in host-native endianness, and are read in host-native endianness.
Which means there needs to be a match, and our servers currently run little-endian.

It should be quite trivial actually to make reading work for other-endianness, I just didn’t put in the effort so far, but its not much. Not sure if its worth creating a followup task for this?

File: crates/symbolicator/src/endpoints/mod.rs [hunk 3]
```diff
@@ -34,7 +34,7 @@ pub fn create_app(service: Service) -> App {
         .route("/proxy/:path", get(proxy).head(proxy))
         .route("/requests/:request_id", get(requests))
         .route("/symbolicate", post(symbolicate))
-        .route("/minidump", post(minidump))
+        .route("/minidump", post(handle_minidump))
```
Comment (flub) [inline crates/symbolicator/src/endpoints/mod.rs:37]: I'm confused by this.  was this pushed to the right PR?

File: crates/symbolicator/src/services/minidump.rs [hunk 4]
```diff
@@ -0,0 +1,858 @@
...
+//! The header is 16 bytes long and structured as follows:
+//!```ignore
+//!bytes     0                     3 4                     7
+//!         +-----------------------+-----------------------+
+//!         | version number      1 | number of threads     |
+//!         +-----------------------+-----------------------+
+//!         | number of frames      | length of symbol data |
+//!         +-----------------------+-----------------------+
+//!```
+//! The thread list as a whole is aligned to 8 bytes.
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator/src/services/minidump.rs:23]: Yes, crashpads helpers expose a way to specify the alignment of the whole stream, which we use:

https://github.com/getsentry/crashpad/pull/43/files#diff-c21425df1de0c6815424235981f49c2c83e04bce7ae097de3b0085742edec35dR94-R97

**Review by flub — APPROVED** [review]
generally LGTM, i don't think there's anything blocking but please have a look at the comments :)

File: crates/symbolicator/src/services/minidump.rs [hunk 5]
```diff
@@ -0,0 +1,858 @@
...
+        header: &'data RawHeader,
+        threads: &'data [RawThread],
+        frames: &'data [RawFrame],
+        symbol_bytes: &'data [u8],
+    }
+
+    impl<'data> Format<'data> {
+        /// Parse our custom minidump extension binary format
+        ///
+        /// TODO: add a better explanation of the format ;-)
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator/src/services/minidump.rs:214]: Hm, I thought I removed this snippet in another commit :-D

File: crates/symbolicator/src/endpoints/mod.rs (hunk 3 above)
Comment (Swatinem) [reply crates/symbolicator/src/endpoints/mod.rs:37]: I was a bit surprised myself. While rebasing on master, the compiler threw an error at me, so I just fixed that :-D

File: crates/symbolicator/src/services/minidump.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (flub) [reply crates/symbolicator/src/services/minidump.rs:11]: isn't the expectation for this to be used by mobile?  so aren't we going to quickly run into different endianness between the generator and reader?

[the remark this answers is not in the document]
Comment (flub) [reply crates/symbolicator/src/services/minidump.rs:11]: which is another way of saying that i think we should create a followup task :)

File: crates/symbolicator/src/endpoints/mod.rs (hunk 3 above)
Comment (Swatinem) [reply crates/symbolicator/src/endpoints/mod.rs]: Oh okay, turns out the problem was that we added a `minidump` crate which was ambiguous with the local `minidump` sub-module. This only happened since I rebased on master which has the axum refactor.

Comment (flub) [reply crates/symbolicator/src/endpoints/mod.rs]: ah, makes sense!
